crates/script/src/execute.rs

```diff
@@ -384,15 +384,21 @@ impl ExecutedState {
 }
 
 impl PreSimulationState {
-    pub fn show_json(&self) -> Result<()> {
-        let result = &self.execution_result;
+    pub async fn show_json(&self) -> Result<()> {
+        let result = &mut self.execution_result.clone();
+        let decoder = &self.execution_artifacts.decoder;
+
+        for (_, trace) in &mut result.traces {
+            decode_trace_arena(trace, decoder).await;
```

Comment:
I'd prefer to inline the decoder here 
```suggestion
        for (_, trace) in &mut result.traces {
            decode_trace_arena(trace, &self.execution_artifacts.decoder).await;
```